processor/filterprocessor/config.go

```diff
@@ -46,6 +47,14 @@ type Config struct {
 
 	Profiles ProfileFilters `mapstructure:"profiles"`
 
+	MetricConditions []common.ContextConditions `mapstructure:"metric_conditions"`
```

Comment:
We're probably gonna need something similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/config.go#L80, so we can support both formats, basic (flat list of conditions) and advanced configurations (this struct).